lightning/src/ln/outbound_payment.rs

```diff
@@ -1117,21 +1140,21 @@ impl OutboundPayments {
 		if let Some(ev) = pending_retry_ev { pending_events.push(ev); }
 	}
 
-	pub(super) fn abandon_payment(&self, payment_id: PaymentId) -> Option<events::Event> {
-		let mut failed_ev = None;
+	pub(super) fn abandon_payment(&self, payment_id: PaymentId,
+		pending_events: &Mutex<Vec<events::Event>>
+	) {
```

Comment:
Just curious what the reason is for changing `abandon_payment`, `check_retry_payments`, etc. to take in and modify `pending_events` as opposed to how it was before. Is it mainly just because it’s a bit cleaner or something else?

Reply:
It's a bit cleaner because we want to keep as much code out of `channelmanager.rs` as possible, since it's already huge :)